File: algorithm/src/main/java/baekjoon/p04597/Main.java
package baekjoon.p04597;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (true) {
            String input = sc.next();
            if (input.equals("#")) {
                break;
            }

            System.out.println(solve(input));
        }
    }

    private static String solve(String input) {
        int[] cnt = new int[2];
        StringBuilder sb = new StringBuilder(input.length());

        for (int i = 0; i < input.length() - 1; i++) {
            cnt[input.charAt(i) - '0']++;
            sb.append(input.charAt(i));
        }

        char check = input.charAt(input.length() - 1);
        sb.append(last(check, cnt));

        return sb.toString();
    }

    private static char last(char check, int[] cnt) {
        switch (check) {
            case 'e':
                return (char) ('0' + cnt[1] % 2);
            case 'o':
                return (char) ('1' - cnt[1] % 2);
            default:
                return '?';
        }
    }

}

File: algorithm/src/main/java/baekjoon/p04597/Main.kt
package baekjoon.p04597

fun main(args: Array<String>) {
    while (true) {
        val input = readLine()!!
        if (input == "#") {
            break
        }

        println(solve(input))
    }
}

private fun solve(input: String): String {
    return buildString {
        append(input.dropLast(1))
        append(bit(input.last(), count { it == '1' }))
    }
}

private fun bit(last: Char, cnt: Int): Char {
    return when (last) {
        'e' -> ('0' + cnt % 2)
        'o' -> ('1' - cnt % 2)
        else -> '?'
    }
}